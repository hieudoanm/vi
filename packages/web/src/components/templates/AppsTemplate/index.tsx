import Head from 'next/head';
import React from 'react';
import Container from '@mui/material/Container';
import PageHeader from '../../molecules/PageHeader';
import Appsbar from '../../organisms/Appsbar';

export type AppsTemplateProps = {
  title?: string;
};

const AppsTemplate: React.FC<AppsTemplateProps> = ({
  title = '',
  children,
}) => {
  return (
    <>
      <Head>
        <title>VI - {title}</title>
      </Head>
      <Appsbar appName="VI" />
      <PageHeader title={title} />
      <Container>
        <div className="py-8">{children}</div>
      </Container>
    </>
  );
};

AppsTemplate.displayName = 'AppsTemplate';
AppsTemplate.defaultProps = { title: '' };

export default AppsTemplate;
