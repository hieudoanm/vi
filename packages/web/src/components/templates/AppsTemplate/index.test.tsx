import { UserProvider } from '@auth0/nextjs-auth0';
import { render } from '@testing-library/react';
import AppsTemplate from '.';

describe('AppsTemplate', () => {
  test('render default', () => {
    const wrapper = render(
      <UserProvider>
        <AppsTemplate title="title" />
      </UserProvider>
    );
    expect(wrapper.container).toMatchSnapshot();
  });
});
