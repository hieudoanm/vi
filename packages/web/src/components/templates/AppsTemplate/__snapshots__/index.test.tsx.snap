// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppsTemplate render default 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  />
  <div
    class="py-12 border-b"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="flex justify-between items-center"
      >
        <h1
          class="text-4xl uppercase"
        >
          title
        </h1>
        <div />
      </div>
    </div>
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="py-8"
    />
  </div>
</div>
`;
